org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacCompilationUnitResolver.java

```diff
@@ -568,7 +570,18 @@ public Void visitClass(ClassTree node, Void p) {
 		var fileManager = (JavacFileManager)context.get(JavaFileManager.class);
 		List<JavaFileObject> fileObjects = new ArrayList<>(); // we need an ordered list of them
 		for (var sourceUnit : sourceUnits) {
-			File unitFile = new File(new String(sourceUnit.getFileName()));
+			File unitFile;
+			if (javaProject != null) {
+				// path is relative to the project, make it absolute
+				IResource asResource = javaProject.getProject().getParent().findMember(new String(sourceUnit.getFileName()));
```

Comment:
If sourceUnit.getFileName() returns a project-relative path like `src/module-info.java`, how can this really work? The project's parent is the IWorkspaceRoot. 

Would it be better to get the project's location and append the project-relative path instead?

Reply:
I fixed the comment instead of removing it